src/order.jl

```diff
@@ -219,15 +219,28 @@ function vertices(
     g::BipartiteGraph, ::Val{side}, ::DynamicDegreeBasedOrder{degtype,direction}
 ) where {side,degtype,direction}
     other_side = 3 - side
+    # compute dist-2 degrees in an optimized way
+    n = nb_vertices(g, Val(side))
+    degrees_dist2 = zeros(Int, n)
+    dist2_neighbor = falses(n)
+    for v in vertices(g, Val(side))
+        fill!(dist2_neighbor, false)
+        for w1 in neighbors(g, Val(side), v)
+            for w2 in neighbors(g, Val(other_side), w1)
+                dist2_neighbor[w2] = true
+            end
+        end
+        degrees_dist2[v] = sum(dist2_neighbor)
+    end
     if degree_increasing(; degtype, direction)
-        degrees = zeros(Int, nb_vertices(g, Val(side)))
+        degrees = zeros(Int, n)
     else
-        degrees = [degree_dist2(g, Val(side), v) for v in vertices(g, Val(side))]  # TODO: optimize
```

Comment:
Do we still use the function `degree_dist2` somewhere?

Reply:
only in the test suite, but it doesn't hurt keeping it (coverage hasn't dropped)